tests/test_format_detection.py

```diff
@@ -202,6 +203,14 @@ def detect(context):
         self.assertEqual(result.exception.failed_alternatives,
             ('bad alternative 1', 'bad alternative 2'))
 
+    @mark_requirement(Requirements.DATUM_GENERAL_REQ)
+    def test_raise_unsupported(self):
+        def detect(context):
+            context.raise_unsupported()
```

Comment:
It looks like it should be the default implementation, shouldn't it?

Reply:
I considered this, but I think the current default implementation (falling back to `find_sources`) is more useful.

Reply:
When it's supposed to be called then?

Reply:
I have added calls to it in this PR.

Reply:
Ok, I expect it to be redesigned quite soon though.